androidTest/java/com/owncloud/android/ui/activity/PublicShareActivityTest.java

```diff
@@ -0,0 +1,1048 @@
+                    testPassword = arguments.getString("TEST_PASSWORD");
+                    servertype = ServerType.fromValue(Integer.parseInt(arguments.getString("TRUSTED")));
+
+                    //Add an account to the device in order to avoid login view
+                    AccountsManager.addAccount(targetContext, testServerURL, testUser, testPassword);
+
+                }
+            };
+
+
+    @Before
+    public void init() {
+
+        // UiDevice available form API level 17
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
+            UiDevice uiDevice =
+                    UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+            try {
+                if (!uiDevice.isScreenOn())
+                    uiDevice.wakeUp();
+            } catch (RemoteException e) {
+                e.printStackTrace();
+            }
+        }
+
+        try {
+            // From API level 23, permissions have to be accepted explicitly if they haven't before
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M &&
+                    grantedPermission() != PackageManager.PERMISSION_GRANTED) {
+
+                //Accept to allow app to manage files on the device
+                UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+                        .findObject(new UiSelector().clickable(true).index(GRANT_BUTTON_INDEX)).click();
+
+                SystemClock.sleep(WAIT_CERTIF_MS);
+                //Accept the untrusted certificate
+                if (servertype == ServerType.HTTPS_NON_SECURE) {
+                    onView(withId(R.id.ok)).perform(click());
+                }
+            }
+
+        } catch (UiObjectNotFoundException e) {
+            e.printStackTrace();
+        }
+
+        //Get Server Capabilities
+        capabilities = AccountsManager.getCapabilities(testServerURL, testUser, testPassword);
+
+        mActivityRule.getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
+    }
+
+
+    @BeforeClass
+    public static void before(){
+        //Needed to use clipboard
+        Looper.prepare();
+    }
+
+    /**
+     *  TEST CASE: Share public a folder (default options)
+     *  PASSED IF: Link created and visible in share view (message of "no links" does not appear)
+     */
+    @Test
+    public void test_01_create_public_link_folder_defaults()
+            throws InterruptedException, NoSuchFieldException, IllegalArgumentException, IllegalAccessException {
+
+        Log_OC.i(LOG_TAG, "Test Share Public Defaults Start");
+        SystemClock.sleep(WAIT_INITIAL_MS);
+
+        //Select share option
+        selectShare(folder);
+
+        //Check that no links are already created
+        onView(withId(R.id.shareNoPublicLinks)).check(matches(isDisplayed()));
+
+        //Depending the server version, send a name or not.
+        if (capabilities.getVersionMayor() >= VERSION_10) {
```

Comment:
Might be better to avoid the check of capabilities and, instead, check if the field to enter the name is visible in the UI or not. If is there, then the test writes it. If it is not, the test doesn't write it.

Let's discuss it for other iteration later.